server/src/main/java/org/opensearch/cluster/node/DiscoveryNode.java

```diff
@@ -509,7 +510,8 @@ public boolean isSearchNode() {
      * @return true if the node contains remote store node attributes, false otherwise
      */
     public boolean isRemoteStoreNode() {
-        return this.getAttributes().keySet().stream().anyMatch(key -> key.startsWith(REMOTE_STORE_NODE_ATTRIBUTE_KEY_PREFIX));
+        return this.getAttributes().keySet().stream().anyMatch(key -> key.equals(REMOTE_STORE_CLUSTER_STATE_REPOSITORY_NAME_ATTRIBUTE_KEY))
+            && this.getAttributes().keySet().stream().anyMatch(key -> key.equals(REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY));
```

Comment:
why do you need to change this?

Reply:
When node has only remote publication / state enabled, the existing check evaluates to true(always) since the repositories share the same prefix `REMOTE_STORE_NODE_ATTRIBUTE_KEY_PREFIX `

This change introduces specific check on the presence of State & Segment repository attributes on node to consider it as RemoteStore node. This is to differentiate RemotePublication node (which may / may not have segment repo)  vs RemoteStore (which will segment repo).

Reply:
ok but now that you mention, i hope it doesnt break assumptions anywhere else.